cli/src/complete.rs

```diff
@@ -43,6 +43,34 @@ pub fn local_bookmarks() -> Vec<CompletionCandidate> {
     })
 }
 
+pub fn new_remote() -> Vec<CompletionCandidate> {
+    with_jj(|mut jj| {
+        let output = jj
+            .arg("git")
+            .arg("remote")
+            .arg("list")
+            .output()
+            .map_err(user_error)?;
+
+        let stdout = String::from_utf8_lossy(&output.stdout);
+
+        let existing_remotes: Vec<_> = stdout
+            .lines()
+            .map(|line| {
+                line.split_once(' ')
+                    .map(|(name, _url)| name)
+                    .unwrap_or_default()
+            })
+            .collect();
+
+        Ok(["origin", "upstream"]
```

Comment:
I personally don't think it's useful to suggest these names, and I wouldn't even mind if they contained existing remotes.

Just use static list if that's useful?

Reply:
You're right, a static list is more than enough.

Reply:
> I personally don't think it's useful to suggest these names, and I wouldn't even mind if they contained existing remotes.

Personally, I feel completions won't be very useful if they're going to suggest things that don't actually work. I'm not very convinced about completions for new remote names. `origin` is already the default remote name, so there's not much value suggesting that. Perhaps `upstream` is slightly more valuable.

If we do want to add completions for `jj git remote add`, I do think we should filter out existing remote names, because providing a completion which causes the resulting command to fail doesn't add any value to the user. In that case we might as well not provide completions.

Reply:
I'm fine with either. It's not a high-stages situation.

The reason I came up with it is because I often create little throw-away repos locally to try something out. These don't have a remote to start. But then I might change my mind and want to add a remote anyway. So I ended up typing "origin" a bunch of times, which isn't a big deal, that's just where the idea came from.

Reply:
@yuja are you OK with keeping the logic of not suggesting existing remote names for new remotes? It's just a 10 line function at this point.

Reply:
My vote (in order of preference):
1. no completion for new remote names
2. simple static list of "common" names
3. dynamic list of "common" names that doesn't exist

In any case, I don't strongly oppose to adding (3) if people like it.

Reply:
I like (3) :slightly_smiling_face:

Reply:
I vote for (1). I'm also fine with (3) if that's the consensus.

Reply:
Alright, I implemented (1).